hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/s3/awssdk/v2/AbstractS3SDKV2Tests.java

```diff
@@ -0,0 +1,288 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import javax.xml.bind.DatatypeConverter;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.ozone.MiniOzoneCluster;
+import org.apache.ozone.test.OzoneTestBase;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.io.TempDir;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CompleteMultipartUploadResponse;
+import software.amazon.awssdk.services.s3.model.CompletedMultipartUpload;
+import software.amazon.awssdk.services.s3.model.CompletedPart;
+import software.amazon.awssdk.services.s3.model.CopyObjectRequest;
+import software.amazon.awssdk.services.s3.model.CopyObjectResponse;
+import software.amazon.awssdk.services.s3.model.CreateMultipartUploadResponse;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.HeadObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+import software.amazon.awssdk.services.s3.model.Tag;
+import software.amazon.awssdk.services.s3.model.Tagging;
+import software.amazon.awssdk.services.s3.model.UploadPartRequest;
+import software.amazon.awssdk.services.s3.model.UploadPartResponse;
+
+/**
+ * This is an abstract class to test the AWS Java S3 SDK operations.
+ * This class should be extended for OM standalone and OM HA (Ratis) cluster setup.
+ *
+ * The test scenarios are adapted from
+ * - https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/javav2/example_code/s3/src/main/java/com/example/s3
+ * - https://github.com/ceph/s3-tests
+ *
+ * TODO: Add tests to different types of S3 client (Async client, CRT-based client)
+ *  See:
+ *   - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/examples-s3.html
+ *   - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/asynchronous.html
+ *   - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/crt-based-s3-client.html
+ */
+@TestMethodOrder(MethodOrderer.MethodName.class)
+public abstract class AbstractS3SDKV2Tests extends OzoneTestBase {
+
+  private static MiniOzoneCluster cluster = null;
+  private static S3Client s3Client = null;
+
+  /**
+   * Create a MiniOzoneCluster with S3G enabled for testing.
+   * @param conf Configurations to start the cluster
+   * @throws Exception exception thrown when waiting for the cluster to be ready.
+   */
+  static void startCluster(OzoneConfiguration conf) throws Exception {
+    cluster = MiniOzoneCluster.newBuilder(conf)
+        .includeS3G(true)
+        .setNumDatanodes(5)
+        .build();
+    cluster.waitForClusterToBeReady();
+    s3Client = cluster.newS3ClientV2();
```

Comment:
v1 client object does not have a close() method, but v2 s3 client has a close() and its tutorial document also closes the object at the end: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html#get-started-code
So let's call close() on it. I think that's a good hygiene.

Reply:
Thanks for noticing this. Added.